### PR TITLE
Rename tag class to allergy class

### LINKED ISSUE #83: Rename tag class to allergy class
Convert existing tag class to allergy class.
Potentially convert t/ to al/ or add constraints for allergy class.

### PR DESCRIPTION
**Pull request by hweinian**
Fixes #83 

Separated tag class with add command, making it a stand-alone function.
The tag command is known as addAllergyCommand.

command format: addAllergy i/VALID NRIC t/VALID TAG

Will add more test case in the future.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddAllergyCommand.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    public static final String COMMAND_WORD = "addAllergy";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds allergy to the patient in the address book. "
+            + "Parameters: NRIC (must be a valid NRIC in the system) "
+            + "[" + PREFIX_TAG + "Allergy]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NRIC + "S1234567A "
+            + PREFIX_TAG + "Insulin "
+            + PREFIX_TAG + "Diabetes";
```
Comment (PresleyChew) [inline src/main/java/seedu/address/logic/commands/AddAllergyCommand.java:31]: "Diabetes" To be removed

File: src/main/java/seedu/address/logic/parser/AddAllergyCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if (!arePrefixesPresent(argMultimap, PREFIX_NRIC, PREFIX_TAG)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddAllergyCommand.MESSAGE_USAGE));
+        }
+        try {
+            Nric nric = ParserUtil.parseNric(argMultimap.getValue(PREFIX_NRIC).get());
+            Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
+
+            logger.info("Successfully parsed nric for AddAllergyCommand: " + nric);
+            logger.info("Successfully parsed tags for AddAllergyCommand: " + tagList);
```
Comment (PresleyChew) [inline src/main/java/seedu/address/logic/parser/AddAllergyCommandParser.java:41]: Nitpicking but perhaps rename tagList to allergies instead

File: src/test/java/seedu/address/logic/commands/AddAllergyCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+        // different types -> returns false
+        assertFalse(addAliceAllergyCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(addAliceAllergyCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(addAliceAllergyCommand.equals(addBobCommand));
+    }
```
Comment (PresleyChew) [inline src/test/java/seedu/address/logic/commands/AddAllergyCommandTest.java:85]: Comprehensive test cases

**Review by PresleyChew — COMMENTED** [review]
Overall LGTM, just nitpicking on the naming.

File: src/test/java/seedu/address/logic/parser/AddAllergyCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void parse_missingFields_failure() {
+        Nric aliceNric = ALICE.getNric();
+        String allergy = "Peanut allergy";
+        Set<Tag> tags = new HashSet<>();
+        tags.add(new Tag(allergy));
+        AddAllergyCommand expectedCommand = new AddAllergyCommand(aliceNric, tags);
+        assertParseFailure(parser, " " + PREFIX_TAG + allergy,
+                String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, AddAllergyCommand.MESSAGE_USAGE));
+    }
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/logic/parser/AddAllergyCommandParserTest.java:43]: Test cases are appropriate and comprehensive 👍🏻

**Review by yong-zhuo — APPROVED** [review]
LGTM overall

**Pull request merged**